include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -109,25 +110,43 @@ inline auto mean(Container const & v)
     return mean(std::execution::seq, v.cbegin(), v.cend());
 }
 
-template<class ForwardIterator>
-auto variance(ForwardIterator first, ForwardIterator last)
+template<class ExecutionPolicy, class ForwardIterator>
+auto variance(ExecutionPolicy&& exec, ForwardIterator first, ForwardIterator last)
 {
     using Real = typename std::iterator_traits<ForwardIterator>::value_type;
     BOOST_ASSERT_MSG(first != last, "At least one sample is required to compute mean and variance.");
     // Higham, Accuracy and Stability, equation 1.6a and 1.6b:
-    if constexpr (std::is_integral<Real>::value)
+    if constexpr (std::is_integral_v<Real>)
     {
-        double M = *first;
-        double Q = 0;
-        double k = 2;
-        for (auto it = std::next(first); it != last; ++it)
+        if constexpr (std::is_same_v<std::remove_reference_t<decltype(exec)>, decltype(std::execution::seq)>)
         {
-            double tmp = *it - M;
-            Q = Q + ((k-1)*tmp*tmp)/k;
-            M = M + tmp/k;
-            k += 1;
+            double M = *first;
+            double Q = 0;
+            double k = 2;
+            for (auto it = std::next(first); it != last; ++it)
+            {
+                double tmp = *it - M;
+                Q = Q + ((k-1)*tmp*tmp)/k;
+                M = M + tmp/k;
+                k += 1;
+            }
+            return Q/(k-1);
+        }
+        else
+        {
+            std::atomic<double> M {*first};
+            std::atomic<double> Q {0};
+            std::atomic<double> k {2};
```

Comment:
I dunno man! Using atomics here is gonna be expensive!

I'd get some performance benchmarks written before you get too far into this.

Also, the Monte-Carlo integration class has an algorithm for pooling variance that you might enjoy looking at.

Reply:
I just wanted to see how this would work with automatic partitioning from `std::for_each`. The answer is not super well.

Reply:
Got it; probably should give you some time before sniping at everything you try!

In any case, looks like I never committed a google benchmark for these functions; do you mind doing that? Should be easy now that you've got the first one done.

Reply:
Can do. I need to validate that the usefulness anyway.